### ISSUE TITLE
Upgrade SecureDrop's Python dependencies

### ISSUE THREAD

**Issue by garrettr** [freedomofpress/securedrop#1301]
We've been blocked on issuing a new release for a long time, mostly due to the increasing support burden for the growing number of SecureDrop installations. As a result, SecureDrop's Python dependencies (which are packaged and installed via a Pip wheel in the `securedrop-app-code` .deb package) are out of date, and should be updated ASAP.

Related: #487, #856.

**Comment by garrettr** [comment]
Neat: https://requires.io/github/freedomofpress/securedrop/requirements/?tag=0.3.6

**Comment by garrettr** [comment]
Looking at that list, I think there are also dependencies in securedrop-requirements.txt that should actually be in test-requirements.txt. Some auditing and testing are required, but I'm pretty sure the following aren't necessary for SecureDrop to run in production:

* beautifulsoup4
* pip-tools
* py

They look like testing dependencies that somehow got into the main requirements file. They should be moved to test-requirements.txt.

**Comment by fowlslegs** [comment]
Progress report:
* I've switched to a new pip-tools based management system based on http://nvie.com/posts/better-package-management/, which I've used to update the requirements files.
* I've been running tests on an app-staging VM with changes as follows:
    * Provision the app-staging VM (and mon-staging VM since you need to do both together)
    * Build the SD app code 0.3.7 package with our new dependencies
    * Copy the SD app code package to the app-staging VM w/ `vagrant scp`
    * Install the new SD app code package on app-staging with `dpkg -i`
    * Run the unit and functional test on app-staging
* The only errors in the unit tests were coming from python-gnupg. I've made some minor changes in the SD app to accomodate for upstream changes.
* The functional tests are failing because Firefox won't start.
    * Selenium 2.5.2
    * Firefox 46.0
    * xvfb 2:1.15.1
    * `xvfb-run firefox` doesn't even work
    * I can't even obtain a log file for it with selenium because it quits instantly. From past experience this seems like a xvfb/display settings error.
    * `echo $DISPLAY` returns `:1` as it should
    *  Xvfb is running on display `:1`
    * @garrettr and I are going to work more on debugging this tomorrow

**Comment by fowlslegs** [comment]
Okay, so I tried installing firefox-esr in the VM and that wouldn't start either. It did, however, give me slightly more output.

```
vagrant@app-staging:/opt/firefox$ xvfb-run ./firefox

(process:5298): GLib-CRITICAL **: g_slice_set_config: assertion 'sys_page_size == 0' failed
Xlib:  extension "RANDR" missing on display ":99".
out of memory
Assertion failure: [unhandlable oom] OOM in createJitRuntime, at /builds/slave/rel-m-esr38-l64_bld-0000000000/build/js/src/jscntxt.cpp:1184
```

Looking up the oom errorr led me to https://forums.grsecurity.net/viewtopic.php?f=3&t=4375. I ran

```
setfattr -n user.pax.flags -vm /usr/lib/firefox/firefox
setfattr -n user.pax.flags -vm /usr/lib/firefox/plugin-container
```

as suggested to disable MPROTECT on FF's plugin-container, and that fixed the problem. All tests are now passing!

**Comment by fowlslegs** [comment]
Closed by #1306.
